brave/brave-core
# Add IPFS public gateway setting

Resolves https://github.com/brave/brave-browser/issues/11563

1st commit: Add a new pref to store public gateway address and update util functions.
2nd commit: [UI change] Add a new setting for customizing the public gateway address.
3rd commit: update tests
4th commit: npm run format

Although github automatically ask for owner reviews, I don't think this PR needs that from net owner and patch-reviewers since it's just updating code in net for a function signature change in `components/ipfs`, and adding new string entries into `chromium_src/chrome/browser/ui/webui/settings/settings_localized_strings_provider.cc`

![ipfs_gateway](https://user-images.githubusercontent.com/4730197/101550277-ddc2aa80-3963-11eb-9ac3-5b28a4e4db81.gif)

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.

## Test Plan:
1. Go to brave://settings and set IPFS resolve method settings to Gateway.
2. Set IPFS public gateway address to https://cf-ipfs.com and verify it is shown in settings.
3. Visit ipfs://QmbWqxBEKC3P8tqsKc98xmWNzrzDtRLMiMPL8wBuTGsMnR
4. URL bar should be shown as https://bafybeigdyrzt5sfp7udm7hu76uh7y26nf3efuylqabf3oclgtqy55fbzdi.pfs.cf-ipfs.com
(See the gif above too for reference)
